lib/AST/Decl.cpp

```diff
@@ -4314,6 +4320,68 @@ const FieldDecl *RecordDecl::findFirstNamedDataMember() const {
   return nullptr;
 }
 
+// Checked C
+
+// Type Parameters
+
+bool RecordDecl::isGeneric() {
+  return IsGeneric;
+}
+
+ArrayRef<TypedefDecl*> RecordDecl::typeParams() {
+  return { TypeParams, NumTypeParams };
+}
+
+void RecordDecl::setTypeParams(const ASTContext& C, ArrayRef<TypedefDecl*> NewTypeParams) {
+  assert(!isGeneric() && "Can't reset type parameters for record");
+  NumTypeParams = NewTypeParams.size();
+  IsGeneric = NumTypeParams > 0;
+  assert(!(isGeneric() && isInstantiated()) && "Record can't be both generic and instantiated");
```

Comment:
Clarification: Can this happen in reality? asserts don't fire on retail bits, so I was wondering if we need to throw an error for this case.

Reply:
This would be a logical error within the compiler - should never happen in reality.